spec/controllers/ansible_repository_controller_spec.rb

```diff
@@ -103,19 +127,93 @@
       end
     end
 
-    context 'refreshing an actual repository' do
-      let(:params) { {:pressed => "ansible_repository_reload"} }
-      let(:display) { "main" }
+    context 'refreshing an actual repository page' do
+      let(:params) { {:action => "button", :pressed => "ansible_repository_reload", :display => display} }
 
       before do
-        controller.instance_variable_set(:@display, display)
+        controller.instance_variable_set(:@breadcrumbs, [])
       end
 
-      it 'calls show and render methods' do
-        expect(controller).to receive(:show)
-        expect(controller).to receive(:render)
-        controller.send(:button)
-        expect(controller.instance_variable_get(:@_params)[:display]).to eq(display)
+      context 'showing repository summary itself' do
+        let(:display) { nil }
+
+        it 'uses @display over params[:display]' do
+          instance_display = 'main'
+          controller.instance_variable_set(:@display, instance_display)
+
+          receives_init_show
+          expect(controller).to receive(:render).with(:update)
+          controller.send(:button)
+
+          expect(controller.instance_variable_get(:@_params)[:display]).to eq(instance_display)
+        end
+
+        it 'renders JS replacing #main_div with show template' do
+          receives_init_show
+          renders_with_template('show')
+          controller.send(:button)
+        end
+      end
+
+      context 'showing output capture subpage' do
+        let(:display) { 'output' }
+
+        it 'calls show_output method' do
+          receives_init_show
+          expect(controller).to receive(:show_output)
+          renders
+          controller.send(:button)
+        end
+
+        it 'renders JS replacing #main_div with show_output template' do
+          receives_init_show
+          renders_with_template('show_output')
+          controller.send(:button)
+        end
+      end
+
+      context 'showing playbooks subpage' do
+        let(:display) { 'playbooks' }
+
+        it 'calls display_playbooks method' do
+          receives_init_show
+          expect(controller).to receive(:display_playbooks)
+          renders
+          controller.send(:button)
+        end
+
+        it 'renders JS replacing #main_div with nested list' do
+          receives_init_show
+          expect(controller).to receive(:nested_list).with(
+            ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook,
+            any_args
+          )
+          renders_with_template('show')
+          controller.send(:button)
+        end
+      end
+
+      def receives_init_show
```

Comment:
Cc @skateman , I think you may have some comments here..

Reply:
@Glutexo methods in a spec are evil, can you rewrite them using shared examples and shared contexts?

Reply:
@himdel @skateman No problem, I’ll rewrite it. Would you please provide me with a link to some resource explaining the evilness of methods in specs?

Reply:
TL;DR: mixing two patterns (rspec ruby vs pure ruby) is never a good idea
If you want a longer explanation, I can try next week if that's okay for you :wink: